test/performance/range/views/view_comparison_seq_read_benchmark.cpp

```diff
@@ -0,0 +1,237 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <list>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/views/all.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+
+enum tag
+{
+    baseline,
+    slice,
+    comp,
+    rev,
+    rev_comp,
+    to_char,
+    to_upper,
+    filter,
+    pseudofilter,
+    until,
+    until_throw
+};
+
+// ============================================================================
+//  SeqAn3
+// ============================================================================
+
+template <typename rng_t, typename counter_t = size_t>
+void sequential_read_impl(benchmark::State & state, rng_t && range, counter_t && counter = 0)
+{
+    using alphabet_t = std::ranges::range_value_t<rng_t>;
+    alphabet_t a;
+
+    for (auto _ : state)
+    {
+        for (auto && c : range)
+            benchmark::DoNotOptimize(a = c);
+
+        counter = 0;
+    }
+}
+
+template <template <typename> typename container_t, tag id>
+void sequential_read(benchmark::State & state)
+{
+    auto rando = seqan3::test::generate_sequence<seqan3::dna4>(10000, 0, 0);
+
+    container_t<seqan3::dna4> seq(rando.begin(), rando.end());
+
+    if constexpr (id == baseline)
+    {
+        sequential_read_impl(state, seq);
+    }
+    else if constexpr (id == slice)
+    {
+        auto long_rando = seqan3::test::generate_sequence<seqan3::dna4>(30000, 0, 0);
```

Comment:
long_rando = 🐕 ?

Reply:
🤣 

I agree 100% that this particular name is badly chosen, and I will definitely change it.